Update permission alerts

alfogrillo opened the pull request:
This PR update system and custom alert for asking permission for the microphone and location.

| First time asking permission | Asking the permission again |
| --- | --- |
| ![mic-first](https://github.com/vector-im/element-x-ios/assets/19324622/3eb0bac9-a42b-4712-91cc-1fddedc2613c)|![mic-later](https://github.com/vector-im/element-x-ios/assets/19324622/23ac28d4-e61b-413b-841b-c9f8ef548c61)|
|![location-first](https://github.com/vector-im/element-x-ios/assets/19324622/7c3188ce-96d1-4aa0-aac9-f89d6be19845)|![location-later](https://github.com/vector-im/element-x-ios/assets/19324622/d4b0aef1-ea33-4408-aa97-66bb508de8f1)|

callumu commented:
A few comments:

- Can we use black as the text for all alerts instead of the system blue?
- Can you update the copy of the alert with the map [to this](https://www.figma.com/file/0MMNu7cTOzLOlWb7ctTkv3/Element-X?type=design&node-id=13665%3A29230&mode=design&t=OhDPeO5zlHazekA2-1)
- I've also changed all instances of 'provide' to 'grant' as I feel like it's better English for the use case. Can you update across all four?

alfogrillo commented:
> * Can we use black as the text for all alerts instead of the system blue?

Nope. These are owned by iOS. We **can't** change the alert title (in bold) as well.

> * Can you update the copy of the alert with the map [to this](https://www.figma.com/file/0MMNu7cTOzLOlWb7ctTkv3/Element-X?type=design&node-id=13665%3A29230&mode=design&t=OhDPeO5zlHazekA2-1)

Done. I updated the screenshots in this PR as well.

> * I've also changed all instances of 'provide' to 'grant' as I feel like it's better English for the use case. Can you update across all four?

Done. I updated the screenshots in this PR as well.

callumu commented:
Great, thanks. This all looks good now
